newsfragments/146.bugfix.rst

```diff
@@ -0,0 +1 @@
+In 0.24.0 ``trio`` introduced `~trio.testing.RaisesGroup` in preparation for changing the default of ``strict_exception_groups`` to ``True``. 0.25.0 made this the new default which broke a number of tests that relied on ``pytest.raises`` - this uplifts the tests to using `~trio.testing.RaisesGroup`.
```

Comment:
This isn't a user-visible change (it only affected the tests), so I think it should be a `misc` ("Miscellaneous internal changes") type newsfragment.

The thing about `RaisesGroup` is not really relevant to what you fixed (you used it in the implementation but it's not the basis of what broke), and mentioning trio 0.24.0 is confusing when this is a change prompted by 0.25.0. I think this can be just "Updated test suite to cope with Trio 0.25.0 and later defaulting `strict_exception_groups` to True. Trio 0.25.0 is now required to run the tests, although trio-asyncio itself still supports older versions."